src/seedu/addressbook/data/person/Name.java

```diff
@@ -61,4 +63,48 @@ public int hashCode() {
         return fullName.hashCode();
     }
 
+    /**
+     * Returns true of the other name is very similar to this name.
+     * Two names are considered similar if ...
+     */
+    public boolean isSimilar(Name other) { 
+        return LevenshteinDistance(fullName, fullName.length(), 
+                                    other.fullName, other.fullName.length()) < SIMILARITY_THRESHOLD;
+    }
+    
+    /**
+     * The "Edit" distance between names. How many characters do I need to substitute 
+     * to get from one name to another
+     * Algorithm from https://en.wikipedia.org/wiki/Levenshtein_distance
+     * @param thisName
+     * @param thatName
+     * @return
+     */
+    private int LevenshteinDistance(String thisName, int thisLength, String thatName, int thatLength) { 
+        int cost;
+
+        /* base case: empty strings */
+        if (thisLength == 0) return thatLength;
```

Comment:
coding standard violation :-1:  you need to use braces even for single line conditionals